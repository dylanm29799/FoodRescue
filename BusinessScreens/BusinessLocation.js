//with help From Expo Documentation https://docs.expo.io/versions/latest/sdk/location/#usage

import React from "react";
import { Platform, Text, View, StyleSheet } from "react-native";
import * as Location from "expo-location";
import Colour from "../constants/Colour";
import { scale } from "../components/ResponsiveText";
import BusinessMap from "../components/businessMap";
import { TouchableOpacity } from "react-native-gesture-handler";
import * as firebase from "firebase";
import "firebase/firestore";

const BusinessLocation = (props) => {
  const dbconnection = firebase.firestore();
  var user = firebase.auth().currentUser;
  var uid = " ";
  try {
    uid = user.uid;
  } catch (err) {
    console.log(err);
  }
  var docRef = dbconnection.collection("businessDetails").doc(uid);
  let longitude = null;
  let latitude = null;
  [(longitude = global.longitude)];
  [(latitude = global.latitude)];
  [console.log(global.latitude, global.longitude)];
  const long = longitude;
  const lat = latitude;
  console.log(long, lat);

  correctLocation = () => {
    docRef.get().then(function (doc) {
      if (doc.exists) {
        console.log("Your details have been updated.");
        return docRef
          .update({
            longitude: longitude,
            latitude: latitude,
          })
          .then(function () {
            console.log("Document successfully updated!");
            props.navigation.navigate({ routeName: "BusinessHome" });
          })
          .catch(function (error) {
            // The document probably doesn't exist.
            console.error("Error updating document: ", error);
          });
      }
    });
  };

  return (
    <View style={styles.container}>
      <BusinessMap style={styles.Map} />

      <View style={styles.textOverMap}>
        <TouchableOpacity style={styles.touchable} onPress={correctLocation}>
          <Text style={styles.text}>This is Your Location?</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.touchable}
          onPress={() =>
            props.navigation.navigate({ routeName: "CorrectLocation" })
          }
        >
          <Text style={styles.text}>This is Not your Location?</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },

  paragraph: {
    flex: 1,
  },

  Map: {
    width: "100%",
    height: "100%",
  },
  textOverMap: {
    position: "absolute",
    flexDirection: "row",
    width: "100%",
    justifyContent: "center",
    alignItems: "center",

    marginTop: scale(20),
  },
  touch: { flex: 1, marginHorizontal: scale(5) },
  text: {
    fontSize: scale(20),
    borderColor: Colour.primaryColour,
    borderWidth: scale(2),
    width: scale(150),
    backgroundColor: "#fff",
    borderRadius: scale(50),
    height: scale(100),
    textAlign: "center",
    paddingTop: scale(30),
    marginHorizontal: scale(5),
  },
});

export default BusinessLocation;
