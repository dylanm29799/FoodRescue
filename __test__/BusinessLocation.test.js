// __tests__/BusinessLocation.test.js

import React from "react";
import BusinessLocation from "../BusinessScreens/BusinessLocation";
import renderer from "react-test-renderer";
import { firebaseConfig } from "../config";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);
test("renders correctly", () => {
  const tree = renderer.create(<BusinessLocation />).toJSON();
  expect(tree).toMatchSnapshot();
});
