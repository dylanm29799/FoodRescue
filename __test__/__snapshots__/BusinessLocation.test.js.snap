// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <AIRMap
      handlePanDrag={false}
      initialRegion={null}
      onChange={[Function]}
      onLayout={[Function]}
      onMapReady={[Function]}
      onMarkerPress={[Function]}
      region={
        Object {
          "latitude": 53.34900146651947,
          "latitudeDelta": 0.02,
          "longitude": -6.243367195129395,
          "longitudeDelta": 0,
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <AIRMapMarker
        coordinate={
          Object {
            "latitude": 53.34900146651947,
            "longitude": -6.243367195129395,
          }
        }
        onPress={[Function]}
        pinColor="#c44536"
        stopPropagation={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
            },
            undefined,
          ]
        }
        title="Your Location"
      />
    </AIRMap>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 42.857142857142854,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#c44536",
              "borderRadius": 107.14285714285714,
              "borderWidth": 4.285714285714286,
              "fontSize": 42.857142857142854,
              "height": 214.28571428571428,
              "marginHorizontal": 10.714285714285714,
              "paddingTop": 64.28571428571428,
              "textAlign": "center",
              "width": 321.42857142857144,
            }
          }
        >
          This is Your Location?
        </Text>
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#c44536",
              "borderRadius": 107.14285714285714,
              "borderWidth": 4.285714285714286,
              "fontSize": 42.857142857142854,
              "height": 214.28571428571428,
              "marginHorizontal": 10.714285714285714,
              "paddingTop": 64.28571428571428,
              "textAlign": "center",
              "width": 321.42857142857144,
            }
          }
        >
          This is Not your Location?
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
